api/v1alpha1/argocd_types.go

```diff
@@ -541,6 +541,27 @@ type ArgoCDServerServiceSpec struct {
 	Type corev1.ServiceType `json:"type"`
 }
 
+// Resource Customization for custom health check
+type ResourceHealthCheck struct {
+	Group string `json:"group"`
+	Kind  string `json:"kind"`
+	Check string `json:"check"`
+}
+
+// Resource Customization for ignore difference
+type ResourceIgnoreDifference struct {
+	Group             string `json:"group"`
+	Kind              string `json:"kind"`
+	JqPathExpressions string `json:"JqPathExpressions"`
```

Comment:
Should `JqPathExpressions` be a `[]string`?

Reply:
yeah probably should be

Reply:
Actually, I think we just take whatever is specified by the user as a single string and put it in the config map because that is how it is documented upstream as well since it is all behind a `|`
```
data:
  resource.customizations.ignoreDifferences.admissionregistration.k8s.io_MutatingWebhookConfiguration: |
    jqPathExpressions:
    - '.webhooks[]?.clientConfig.caBundle'
```

so this probably doesn't need to be an array, user is responsible for specifying how many ever expressions they want to